packages/bson-transpilers/test/yaml/native.yaml

```diff
@@ -1514,6 +1604,7 @@ tests:
           java: "Pattern.compile(\"(?:)\")"
           csharp: "new Regex(\"(?:)\")"
           object: "RegExp('')"
+          php: "'/(?:)/'"
```

Comment:
What is the use case for this template? PHP doesn't have a native regex type, so I don't see what the point of yielding a string here would be.

If this is meant to serialize to a BSON type, then we should consider [MongoDB\BSON\Regex](https://www.php.net/manual/en/class.mongodb-bson-regex.php)).

Reply:
There are two RegExp types for the internal presentation: common RegExp and BSONRegExp. I've made the first one as the converting into common string with RegExp expression, which can be used in functions like preg_match or preg_replace or others. The BSONRegExp template uses \MongoDB\BSON\Regex.

Reply:
I think my problem is not understanding how the RegExp template is used. I didn't see anything in `templates.yaml` referring to PHP's `preg` API, so I'm curious how these strings are going to be used by the user.

Presumably BSONRegExp is used to preserve a regex BSON type (e.g. inserting a document with a regex field), which makes sense, but what is the context in which a string for the RegExp template might appear? Does Compass use this in conjunction with a `$regex` query operator?

Reply:
Fixed. All native RegExps were changed to MongoDB\BSON\Regex.

Reply:
Noted. OK to resolve.